Geometry/CSCGeometryBuilder/src/CSCGeometryParsFromDD.cc

```diff
@@ -426,24 +453,31 @@ bool CSCGeometryParsFromDD::build(const cms::DDCompactView* cview,
       }
 
       auto wirespacing = fv.get<double>("WireSpacing");
-      wg.wireSpacing = static_cast<double>(wirespacing);
+      wg.wireSpacing = static_cast<double>(wirespacing / dd4hep::cm);
+      edm::LogVerbatim("CSCGeometryParsFromDD") << "(2) wireSpacing: " << wg.wireSpacing / dd4hep::cm;
 
       auto alignmentpintofirstwire = fv.get<double>("AlignmentPinToFirstWire");
-      wg.alignmentPinToFirstWire = static_cast<double>(alignmentpintofirstwire);
+      wg.alignmentPinToFirstWire = static_cast<double>(alignmentpintofirstwire / dd4hep::cm);
+      edm::LogVerbatim("CSCGeometryParsFromDD")
+          << "(3) alignmentPinToFirstWire: " << wg.alignmentPinToFirstWire / dd4hep::cm;
 
       auto totnumwiregroups = fv.get<double>("TotNumWireGroups");
       wg.numberOfGroups = static_cast<int>(totnumwiregroups);
 
       auto lengthoffirstwire = fv.get<double>("LengthOfFirstWire");
-      wg.narrowWidthOfWirePlane = static_cast<double>(lengthoffirstwire);
+      wg.narrowWidthOfWirePlane = static_cast<double>(lengthoffirstwire / dd4hep::cm);
+      edm::LogVerbatim("CSCGeometryParsFromDD")
+          << "(4) narrowWidthOfWirePlane: " << wg.narrowWidthOfWirePlane / dd4hep::cm;
 
       auto lengthoflastwire = fv.get<double>("LengthOfLastWire");
-      wg.wideWidthOfWirePlane = static_cast<double>(lengthoflastwire);
+      wg.wideWidthOfWirePlane = static_cast<double>(lengthoflastwire / dd4hep::cm);
+      edm::LogVerbatim("CSCGeometryParsFromDD") << "(5) wideWidthOfWirePlane: " << wg.wideWidthOfWirePlane / dd4hep::cm;
 
       auto radialextentofwireplane = fv.get<double>("RadialExtentOfWirePlane");
-      wg.lengthOfWirePlane = static_cast<double>(radialextentofwireplane);
+      wg.lengthOfWirePlane = static_cast<double>(radialextentofwireplane / dd4hep::cm);
+      edm::LogVerbatim("CSCGeometryParsFromDD") << "(6) lengthOfWirePlane: " << wg.lengthOfWirePlane / dd4hep::cm;
 
-      uparvals.emplace_back((wg.wireSpacing) * 10.0);
+      uparvals.emplace_back(wg.wireSpacing / dd4hep::mm);
```

Comment:
We have to be very careful here. I don't think any values that go into the `uparvals` should be converted. They are plain numbers that the CSC code knows how to handle.
I'm also confused about `wireSpacing`. In the old DD code, `wireSpacing` is not changed, so why is it converted here?

Reply:
@cvuosalo : I deleted "/dd4hep::mm"inside uparvals and I set wireSpacing in mm